test/src/test/java/org/corfudb/runtime/object/transactions/OptimisticTransactionBenchmark.java

```diff
@@ -0,0 +1,169 @@
+
+            rt = getNewRuntime();
+
+            // Every map will be initialized with this map
+            Map<Integer, Integer> initialMap = new HashMap<>();
+            for (int i = 0; i < NUM_KEYS; i++) {
+                initialMap.put(i, i);
+            }
+
+            maps = new ArrayList<>();
+            for (int i = 0; i < NUM_MAPS; i++) {
+                maps.add(rt.getObjectsView().build()
+                        .setStreamName(Integer.toString(i))
+                        .setTypeToken(new TypeToken<SMRMap<Integer, Integer>>() {
+                        })
+                        .open());
+                // Initialize the map with initial map
+                maps.get(i).putAll(initialMap);
+            }
+
+        }
+
+        /**
+         * Get the next key.
+         */
+        int getNextKey() {
+            return r.nextInt(NUM_KEYS);
+        }
+
+        /**
+         * Get the next map.
+         */
+        @Nonnull
+        Map<Integer, Integer> getNextMap() {
+            return maps.get(r.nextInt(NUM_MAPS));
+        }
+
+        /**
+         * A simple interface which defines the transaction to be executed.
+         */
+        @FunctionalInterface
+        interface BenchmarkedTransaction {
+            void doTransaction();
+        }
+
+        /**
+         * Do a transaction.
+         */
+        void doTransaction(@Nonnull TransactionCounters tc, @Nonnull BenchmarkedTransaction tx) {
+            while (true) {
+                tc.transactions++;
+                try {
+                    rt.getObjectsView().TXBegin();
+                    tx.doTransaction();
+                    rt.getObjectsView().TXEnd();
+                    return;
+                } catch (TransactionAbortedException tae) {
+                    tc.aborts++;
+                }
+            }
+        }
+    }
+
+    /**
+     * Counters for measuring the number of actual transactions dispatched and how many were aborted.
+     * The {@link OptimisticBenchmarkState#doTransaction(TransactionCounters, OptimisticBenchmarkState.BenchmarkedTransaction)}
+     * method measures goodput, so these counters give us actual transaction and abort rate.
+     */
+    @AuxCounters
+    @State(Scope.Thread)
+    public static class TransactionCounters {
+        public long transactions;
+        public long aborts;
+
+        @Setup(Level.Iteration)
+        public void cleanup() {
+            transactions = 0;
+            aborts = 0;
+        }
+    }
+
+    /**
+     * Measure transaction performance. Each transaction reads a key and write a key (in a different map).
+     * The number of maps, keys and operations per transaction vary depending on the given parameters,
+     * but the read:write ratio is fixed at 1:1.
+     */
+    @Benchmark
+    @Warmup(iterations = 5)
+    @Measurement(iterations = 10)
+    @Threads(5)
+    public void readWriteDiffMap(OptimisticBenchmarkState state, TransactionCounters tc, Blackhole bh) {
```

Comment:
This test looks not accurate, the loops can be used https://hg.openjdk.java.net/code-tools/jmh/file/1c11c886e0c8/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_11_Loops.java